packages/x-components/src/x-modules/url/store/__tests__/actions.spec.ts

```diff
@@ -54,6 +54,67 @@ describe('testing Url module actions', () => {
 
       expect(window.location.search).toEqual('');
     });
+
+    it('should push the state when the parameter is pushable', async () => {
+      window.history.pushState = jest.fn();
+
+      resetUrlStateWith(store, {
+        config: {
+          urlParamNames: {
+            query: 'q',
+            relatedTags: 'tag'
+          }
+        },
+        params: {
+          query: 'sudadera',
+          filters: [],
+          relatedTags: [],
+          page: 1,
+          scroll: 2
+        }
+      });
+
+      await store.dispatch(actionKeys.updateUrl);
+
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(window.history.pushState).toHaveBeenCalled();
```

Comment:
As you said, maybe we can add another test to check replaceSate or we can add it in the first It:

```
    it('should add the correct params to the url', async () => {
       window.history.replaceState = jest.fn();
       .
      .
      expect(window.history.replaceState).toHaveBeenCalled();
```